Tribler/community/gigachannel/community.py

```diff
@@ -0,0 +1,166 @@
+        :type blob: TruncatedChannelPlayloadBlob
+        :returns: None
+        """
+        for truncated_channel in blob.payload_list:
+            # The database stores the long format of the keys
+            longpk = "LibNaCLPK:" + truncated_channel.public_key
+            if truncated_channel.infohash not in self.download_queue:
+                with db_session:
+                    channel = self.tribler_session.lm.mds.ChannelMetadata.get_channel_with_id(longpk)
+                    if not channel:
+                        # Insert a new channel entry into the database.
+                        # We set the version to 0 so that we receive the up-to-date information later.
+                        self.tribler_session.lm.mds.ChannelMetadata.from_dict({
+                            'infohash': truncated_channel.infohash,
+                            'public_key': longpk,
+                            'title': truncated_channel.title,
+                            'version': 0
+                        })
+                        self.download_queue.append(truncated_channel.infohash)
+                    elif truncated_channel.version > channel.local_version:
+                        # The sent version is newer than the one we have, queue the download.
+                        channel.infohash = truncated_channel.infohash
+                        self.download_queue.append(truncated_channel.infohash)
+                    # We don't update anything if the channel version is older than the one we know.
+
+    def update_from_download(self, download):
+        """
+        Given a channel download, update the amount of votes.
+
+        :param download: the channel download to inspect
+        :type download: LibtorrentDownloadImpl
+        :returns: None
+        """
+        infohash = download.tdef.get_infohash()
+        with db_session:
+            channel = self.tribler_session.lm.mds.ChannelMetadata.get_channel_with_infohash(infohash)
+            channel.votes = download.get_num_connected_seeds_peers()[0]
+
+    def download_completed(self, download):
+        """
+        Callback for when a channel download finished.
+
+        :param download: the channel download which completed
+        :type download: LibtorrentDownloadImpl
+        :returns: None
+        """
+        if self.request_cache.has(u"channel-download-cache", 0):
+            self.request_cache.pop(u"channel-download-cache", 0)
+        self.update_from_download(download)
+
+    def update_states(self, states_list):
+        """
+        Callback for when the download states are updated in Tribler.
+        We still need to filter out the channel downloads from this list.
+
+        :param states_list: the list of download states
+        :type states_list: [DownloadState]
+        :returns: None
+        """
+        for ds in states_list:
+            if ds.get_download().dlconfig.get('download_defaults', 'channel_download'):
```

Comment:
You can use `dlconfig.get_channel_download()` instead.

Reply:
I tried, but it's apparently some kind of other runtime configuration copy, which is missing the method.